hr_attendance_calendar/models/hr_attendance_day.py

```diff
@@ -216,24 +216,31 @@ def _compute_extra_hours(self):
 
             att_day.extra_hours = extra_hours * coefficient
 
-# @api.multi
-# @api.depends('extra_hours')
-# def _compute_extra_hours_lost(self):
-#     for att_day in self:
-#         att_days_ids = self.env['hr.attendance.day'].search(
-#             [('employee_id', '=', att_day.employee_id.id)])
-#
-#         total_extra_hours = sum(att_days_ids.mapped('extra_hours'))
-#         total_lost_hours = sum(att_days_ids.mapped('extra_hours_lost'))
-#         max_extra_hours = float(self.env['ir.config_parameter'].get_param(
-#             'hr_attendance_calendar.max_extra_hours'))
-#
-#         difference = (total_extra_hours-total_lost_hours)-max_extra_hours
-#
-#         if difference > 0:
-#             att_day.extra_hours_lost = float(difference)
-#         else:
-#             att_day.extra_hours_lost = 0.0
+    @api.multi
+    def write(self, vals):
+        super(HrAttendanceDay, self).write(vals)
+
+        if 'worked_hours' in vals:
+            for att_day in self:
+                att_days_future = self.search([
+                    ('date', '>=', att_day.date),
+                    ('employee_id', '=', att_day.employee_id.id)])
+                att_days_future.compute_extra_hours_lost()
+
+    @api.multi
+    @api.depends('extra_hours')
+    def compute_extra_hours_lost(self):
+        for att_day in self:
+            data = self.env[
+                'hr_attendance_day.extra_hours_evolution_day_report'].search(
+                [('hr_date', '=', att_day.date),
+                 ('employee_id', '=', att_day.employee_id.id)])
+
+            max_extra_hours = float(self.env['ir.config_parameter'].get_param(
+                         'hr_attendance_calendar.max_extra_hours'))
+
+            if data.balance > max_extra_hours:
```

Comment:
`if data.balance > max_extra_hours > 0:`
to avoid putting extra hours lost when the limit is not set (and is 0)